[WIP][Doctests] Fix rst -> mdx

patrickvonplaten opened the pull request:
# What does this PR do?

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

In this PR, the already doc tests are enabled. The changes to `quickstart.mdx` show the necessary changes to make the doc tests work


## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- albert, bert, xlm: @LysandreJik
- blenderbot, bart, marian, pegasus, encoderdecoder,  t5: @patrickvonplaten, @patil-suraj
- longformer, reformer, transfoxl, xlnet: @patrickvonplaten
- fsmt: @stas00
- funnel: @sgugger
- gpt2: @patrickvonplaten, @LysandreJik
- rag: @patrickvonplaten, @lhoestq
- tensorflow: @LysandreJik

Library:

- benchmarks: @patrickvonplaten
- deepspeed: @stas00
- ray/raytune: @richardliaw, @amogkam
- text generation: @patrickvonplaten
- tokenizers: @n1t0, @LysandreJik
- trainer: @sgugger
- pipelines: @LysandreJik

Documentation: @sgugger

HF projects:

- datasets: [different repo](https://github.com/huggingface/datasets)
- rust tokenizers: [different repo](https://github.com/huggingface/tokenizers)

Examples:

- maintained examples (not research project or legacy): @sgugger, @patil-suraj
- research_projects/bert-loses-patience: @JetRunner
- research_projects/distillation: @VictorSanh

 -->

patrickvonplaten committed:
revive

HuggingFaceDocBuilder commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_15329). All of your documentation changes will be reflected on that endpoint.

patrickvonplaten committed:
up

patrickvonplaten commented on .github/workflows/doctests.yml:
```diff
@@ -4,9 +4,16 @@ on:
-  repository_dispatch:
-  schedule:
-    - cron: "0 0 * * *"
+  pull_request:
```
Will revert this once PR is approved

patrickvonplaten commented:
@LysandreJik @sgugger, It'd be great if you could take a first look here.

There are essentially two things that I'd like to verify/discuss here:
1. The seperation line `===PT-TF-SPLIT===` in Python code ```python is interpreted as an output which then leads the doctest to fail with:
```
Expected: ===PT-TF-SPLIT===
Got: Nothing
```
In this PR I replaced `===PT-TF-SPLIT===` with `>>> # ===PT-TF-SPLIT===` so that the split is considered as a comment by doctests and they pass.

2. The second problem is that if the code snippet ends with ``` directly after the last python command, the doctests also interpret this as an expected output. This is why I added a newline before every ``` in the end which is not really a good idea IMO. I think it might make sense to just a quick preprocessing of all files that we run the doc tests on to add an empty space before the final ``` -> and then run the doc tests. Think this is better than adding a new line before every ``` which also doesn't go well with @sgugger's style_doc.py file.

patrickvonplaten commented on docs/source/quicktour.mdx:
```diff
@@ -59,6 +59,7 @@ pip install tensorflow
 >>> from transformers import pipeline
```
This file as changed right now would pass the doc-tests

patrickvonplaten commented on docs/source/quicktour.mdx:
```diff
@@ -81,6 +84,7 @@ a list of dictionaries like this one:
 ...     print(f"label: {result['label']}, with score: {round(result['score'], 4)}")
 label: POSITIVE, with score: 0.9998
 label: NEGATIVE, with score: 0.5309
+
```
This is what I meant with my 2. point

patrickvonplaten commented on docs/source/quicktour.mdx:
````diff
@@ -113,8 +118,9 @@ We will need two classes for this. The first is [`AutoTokenizer`], which we will
 ```py
 >>> from transformers import AutoTokenizer, AutoModelForSequenceClassification
-===PT-TF-SPLIT===
+>>> # ===PT-TF-SPLIT===
````
This is what I meant with my 1. point

patrickvonplaten commented on utils/documentation_tests.txt:
```diff
@@ -1,2 +1,2 @@
-docs/source/quicktour.rst
-docs/source/task_summary.rst
\ No newline at end of file
+docs/source/quicktour.mdx
```
this one would pass as is now

patrickvonplaten commented on utils/documentation_tests.txt:
```diff
@@ -1,2 +1,2 @@
-docs/source/task_summary.rst
\ No newline at end of file
+docs/source/quicktour.mdx
+docs/source/task_summary.mdx
```
this one doesn't pass as is now

LysandreJik reviewed (commented):
Sorry I got a bit lost, do you still need a review on this one @patrickvonplaten?

patrickvonplaten commented:
Closing this one as it's more or less a duplicate of the one we merged last week and requires the `PT-TF` switch to be solved before. Will open a new one then :-)
